Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -63,6 +65,21 @@ public struct HTTPRequestHead: Equatable {
     }
 }
 
+/// Internal representation of an URI
+enum Uri {
+    case string(String)
+    case byteBuffer(ByteBuffer)
+
+    var asString: String {
```

Comment:
I think the Swift convention here is actually to provide an extension on `String` that gives it an appropriate constructor from this type.